core/op_s3.go

```diff
@@ -202,43 +228,53 @@ func S3Download(job *Job, wp *WorkerParams) (stats.StatType, error) {
 
 	if err != nil {
 		os.Remove(destFn) // Remove partly downloaded file
-	} else if job.opts.Has(opt.DeleteSource) {
-		err = client.Delete(wp.ctx, job.args[0].s3.Bucket, job.args[0].s3.Key)
+		return opType, &JobResponse{status: statusErr, err: err}
 	}
 
-	return opType, err
+	if job.opts.Has(opt.DeleteSource) {
+		err := client.Delete(wp.ctx, job.args[0].s3.Bucket, job.args[0].s3.Key)
+		if err != nil {
+			return opType, &JobResponse{status: statusErr, err: err}
+		}
+	}
+
+	return opType, &JobResponse{status: statusSuccess}
 }
 
-func S3Upload(job *Job, wp *WorkerParams) (stats.StatType, error) {
+func S3Upload(job *Job, wp *WorkerParams) (stats.StatType, *JobResponse) {
 	const opType = stats.S3Op
 
 	if ex, err := job.args[0].Exists(wp); err != nil {
-		return opType, err
+		return opType, &JobResponse{status: statusErr, err: err}
 	} else if !ex {
-		return opType, os.ErrNotExist
+		return opType, &JobResponse{status: statusErr, err: os.ErrNotExist}
```

Comment:
what about a function like

```
func jobResponse(err error) *JobResponse {
	if err == nil {
		return &JobResponse{status: statusSuccess}
	}
	return &JobResponse{status: statusErr, err: err}
}
```

to make it more readable?

Reply:
done